src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -316,20 +334,137 @@ object P2PKScriptPubKey extends Factory[P2PKScriptPubKey] {
 
 }
 
+/**
+  * Represents a scriptPubKey that contains OP_CHECKLOCKTIMEVERIFY.
+  * Adds an absolute/defined locktime condition to any scriptPubKey.
+  * [[https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki]]
+  * Format: <locktime> OP_CLTV OP_DROP <scriptPubKey>
+  */
+trait CLTVScriptPubKey extends ScriptPubKey {
+  /**
+    * Determines the nested ScriptPubKey inside the CLTVScriptPubKey
+    * @return
+    */
+  def scriptPubKeyAfterCLTV : ScriptPubKey = ScriptPubKey(asm.slice(4, asm.length))
+}
+
+object CLTVScriptPubKey extends Factory[CLTVScriptPubKey] {
+  private case class CLTVScriptPubKeyImpl(hex : String) extends CLTVScriptPubKey
+
+  override def fromBytes (bytes : Seq[Byte]) : CLTVScriptPubKey = {
+    val asm = ScriptParser.fromBytes(bytes)
+    CLTVScriptPubKey(asm)
+  }
+  def fromAsm (asm : Seq[ScriptToken]) : CLTVScriptPubKey = {
+    require(isCLTVScriptPubKey(asm), "Given asm was not a CLTVScriptPubKey, got: " + asm)
+    val hex = asm.map(_.hex).mkString
+    CLTVScriptPubKeyImpl(hex)
+  }
+
+  def apply (asm: Seq[ScriptToken]) : CLTVScriptPubKey = fromAsm(asm)
+
+  /**
+    * Creates a P2PKH-formatted CLTVScriptPubKey
+    * @param locktime block height or timestamp
+    * @param pubKey public key corresponding to the CLTV output
+    * @return
+    */
+  def apply(locktime : ScriptNumber, pubKey : ECPublicKey) : CLTVScriptPubKey = {
```

Comment:
Not sure if I like the idea of exposing this function to everyone, they can just call the other `apply` function with an already built `P2PKHScriptPubKey`... let me think about it more